backend/cmd/api/internal/model/model.go

```diff
@@ -5,43 +5,85 @@ package model
 
 import (
 	"context"
+	"log"
 
 	"github.com/CMS-Enterprise/ztmf/backend/internal/db"
 	"github.com/Masterminds/squirrel"
 	"github.com/jackc/pgx/v5"
 )
 
-// sqlBuilder is a convenient way to reference a squirrel.StatementBuilder that
+// stmntBuilder is a convenient way to reference a squirrel.StatementBuilder that
 // uses the PostgreSQL $1,$2,... format of placeholders
-var sqlBuilder = squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar)
+var stmntBuilder = squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar)
+
+// SqlBuilder allows methods to receive different types like squirrel.InsertBuilder, squirrel.UpdateBuilder, etc. that all implement the ToSql method
+type SqlBuilder interface {
+	ToSql() (string, []interface{}, error)
+}
+
+// query is a proxy to *pgx.Conn.Query and wrapper around pgx.CollectRows, enabling the centralizing of event tracking
+func query[T any](ctx context.Context, sqlb SqlBuilder, fn pgx.RowToFunc[T]) ([]T, error) {
```

Comment:
`query` and `queryRow` are perhaps the crux and most crucial piece of this refactor. Every other change across the codebase is just to conform to the change in signature.

Previously, all model functions and struct methods were duplicating efforts when it came to converting the sql builder into a prepared sql statement and bound arguments, and duplicating efforts related to error handling as well.

This change centralizes those steps so that model functions and struct methods need only pass the sql builder and the appropriate pgx.RowToFunc function